core/src/main/java/hudson/logging/LogRecorderManager.java

```diff
@@ -66,9 +67,23 @@
  */
 public class LogRecorderManager extends AbstractModelObject implements ModelObjectWithChildren, StaplerProxy {
     /**
-     * {@link LogRecorder}s keyed by their {@linkplain LogRecorder#name name}.
+     * {@link LogRecorder}s keyed by their {@linkplain LogRecorder#getName()}  name}.
      */
-    public transient final Map<String,LogRecorder> logRecorders = new CopyOnWriteMap.Tree<>();
+    public List<LogRecorder> logRecorders;
```

Comment:
this breaks compatibility for a couple of plugins 😢 
https://github.com/search?q=org%3Ajenkinsci+logRecorders&type=Code

Reply:
![image](https://user-images.githubusercontent.com/1661688/75633255-9ef90a80-5c03-11ea-8412-2a6be5846264.png)

Reply:
anyone got any ideas in how to make this compatible =/

I can readResolve to keep populate the field but plugins expect to be able to add more logrecorders by putting to the map

Reply:
ah think I have a solution 😄

Reply:
Pushed where I got to but for some reason copying the list back to the original map isn't working, if I try lookup the map afterwards it's empty =/